src/python/pants/backend/jvm/tasks/coverage/jacoco.py

```diff
@@ -6,19 +6,50 @@
                         unicode_literals, with_statement)
 
 import functools
+import os
 
+from pants.backend.jvm.subsystems.jvm_tool_mixin import JvmToolMixin
 from pants.backend.jvm.tasks.coverage.engine import CoverageEngine
+from pants.base.exceptions import TaskError
+from pants.java.jar.jar_dependency import JarDependency
 from pants.subsystem.subsystem import Subsystem
+from pants.util import desktop
+from pants.util.dirutil import safe_delete, safe_mkdir
 
 
 class Jacoco(CoverageEngine):
   """Class to run coverage tests with Jacoco."""
 
-  class Factory(Subsystem):
+  class Factory(Subsystem, JvmToolMixin):
     options_scope = 'jacoco'
 
     @classmethod
-    def create(cls, settings, targets, execute_java_for_targets):
+    def register_options(cls, register):
+      super(Jacoco.Factory, cls).register_options(register)
+
+      # We need to inject the jacoco agent at test runtime
+      cls.register_jvm_tool(register,
+                        'jacoco-agent',
+                        classpath=[
+                          JarDependency(
+                            org='org.jacoco',
+                            name='org.jacoco.agent',
+                            rev='0.7.9',
+                            classifier='runtime',
+                            intransitive=True)
+                        ])
+
+      # We need to inject the jacoco agent at test runtime
+      cls.register_jvm_tool(register,
+                        'jacoco-cli',
+                        classpath=[
+                          JarDependency(
+                            org='org.jacoco',
+                            name='org.jacoco.cli',
```

Comment:
The name of the artifact is really `org.jacoco.cli`? That's pretty odd.

Reply:
yeah, i think so: https://oss.sonatype.org/content/repositories/snapshots/org/jacoco/org.jacoco.cli/